kratos/containers/distributed_vector_importer.h

```diff
@@ -0,0 +1,343 @@
+                GetComm().SendRecv(send_buffer, color, 0, recv_buffer, color, 0); //TODO, we know all the sizes, we shall use that!
+
+                //write the recv_data onto the output
+                for(IndexType i=0; i<recv_buffer.size(); ++i)
+                    ImportedData(local_ids[i]) = recv_buffer[i];
+            }
+        }
+        //treat local data
+        for(IndexType i=0; i<mLocallyOwnedIds.size(); ++i )
+
+            ImportedData(mIdOfLocallyOwnedTerms[i]) = data_vector( mLocallyOwnedIds[i] );
+
+        return ImportedData;
+
+    }
+
+    /// Destructor.
+    virtual ~DistributedVectorImporter(){}
+
+    ///@}
+    ///@name Operators
+    ///@{
+    const DataCommunicator& GetComm() const{
+        return mrComm;
+    }
+
+    
+    ///@}
+    ///@name Operations
+    ///@{
+
+    ///@}
+    ///@name Access
+    ///@{
+
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    virtual std::string Info() const
+    {
+    std::stringstream buffer;
+    buffer << "DistributedVectorImporter" ;
+    return buffer.str();
+    }
+
+    /// Print information about this object.
+    virtual void PrintInfo(std::ostream& rOStream) const {rOStream << "DistributedVectorImporter";}
+
+    /// Print object's data.
+    virtual void PrintData(std::ostream& rOStream) const {}
+
+    ///@}
+    ///@name Friends
+    ///@{
+
+
+    ///@}
+
+protected:
+    ///@name Protected static Member Variables
+    ///@{
+
+
+    ///@}
+    ///@name Protected member Variables
+    ///@{
+
+
+    ///@}
+    ///@name Protected Operators
+    ///@{
+
+
+    ///@}
+    ///@name Protected Operations
+    ///@{
+
+
+    ///@}
+    ///@name Protected  Access
+    ///@{
+
+
+    ///@}
+    ///@name Protected Inquiry
+    ///@{
+
+
+    ///@}
+    ///@name Protected LifeCycle
+    ///@{
+
+
+    ///@}
+
+private:
+    ///@name Static Member Variables
+    ///@{
+
+
+    ///@}
+    ///@name Member Variables
+    ///@{
+    const DataCommunicator& mrComm;
+    typename DistributedNumbering<IndexType>::UniquePointer mpNumbering;
+
+
+    IndexType mImportedDataSize;
+    std::unordered_map<int, std::vector<IndexType>> mto_send_by_color;
+    std::unordered_map<MpiIndexType, std::vector<IndexType>> mlocal_i_by_color;
+    std::vector<IndexType> mLocallyOwnedIds;
+    std::vector<IndexType> mIdOfLocallyOwnedTerms;
+    std::vector<int> mvector_comm_colors;
+
+    ///@}
+    ///@name Private Operators
+    ///@{
+
+
+    ///@}
+    ///@name Private Operations
+    ///@{
+
+
+    ///@}
+    ///@name Private  Access
+    ///@{
+
+
+    ///@}
+    ///@name Private Inquiry
+    ///@{
+
+
+    ///@}
+    ///@name Un accessible methods
+    ///@{
+
+    /// Assignment operator.
+    DistributedVectorImporter& operator=(DistributedVectorImporter const& rOther){}
+
+    /// Copy constructor.
+    DistributedVectorImporter(DistributedVectorImporter const& rOther){}
```

Comment:
Why is the copy constructor unaccessible in the importer but usable in the exporter?